data/mods/gen7/random-teams.ts

```diff
@@ -29,568 +30,666 @@ const ZeroAttackHPIVs: {[k: string]: SparseStatsTable} = {
+		this.incompatibleMoves(moves, movePool, ['drainpunch', 'focusblast'], ['closecombat', 'highjumpkick', 'superpower']);
+		this.incompatibleMoves(moves, movePool, 'stoneedge', 'headsmash');
+		this.incompatibleMoves(moves, movePool, 'dracometeor', 'dragonpulse');
+		this.incompatibleMoves(moves, movePool, 'dragonclaw', 'outrage');
+		this.incompatibleMoves(moves, movePool, 'knockoff', ['darkestlariat', 'darkpulse', 'foulplay']);
+		if (!types.includes('Dark') && preferredType !== 'Dark') {
+			this.incompatibleMoves(moves, movePool, 'knockoff', ['pursuit', 'suckerpunch']);
+		}
+
+		// Status move incompatibilities
+		const statusInflictingMoves = ['thunderwave', 'toxic', 'willowisp', 'yawn'];
+		if (!abilities.has('Prankster')) {
+			this.incompatibleMoves(moves, movePool, statusInflictingMoves, statusInflictingMoves);
+		}
+		this.incompatibleMoves(moves, movePool, 'toxic', 'toxicspikes');
+		this.incompatibleMoves(moves, movePool, 'taunt', 'disable');
+		this.incompatibleMoves(moves, movePool, 'defog', ['leechseed', 'substitute']);
+
+		// Assorted hardcodes go here:
+		// Lunatone
+		this.incompatibleMoves(moves, movePool, 'moonlight', 'rockpolish');
+		// Smeargle
+		this.incompatibleMoves(moves, movePool, 'destinybond', 'whirlwind');
+		// Z-Conversion Porygon-Z
+		if (species.id === 'porygonz') {
+			this.incompatibleMoves(moves, movePool, 'shadowball', 'recover');
+		}
+		// Liepard
+		this.incompatibleMoves(moves, movePool, 'copycat', 'uturn');
+		// Seviper
+		this.incompatibleMoves(moves, movePool, 'switcheroo', 'suckerpunch');
+	}
+
+	// Checks for and removes incompatible moves, starting with the first move in movesA.
+	incompatibleMoves(
+		moves: Set<string>,
+		movePool: string[],
+		movesA: string | string[],
+		movesB: string | string[],
+	): void {
+		const moveArrayA = (Array.isArray(movesA)) ? movesA : [movesA];
+		const moveArrayB = (Array.isArray(movesB)) ? movesB : [movesB];
+		if (moves.size + movePool.length <= this.maxMoveCount) return;
+		for (const moveid1 of moves) {
+			if (moveArrayB.includes(moveid1)) {
+				for (const moveid2 of moveArrayA) {
+					if (moveid1 !== moveid2 && movePool.includes(moveid2)) {
+						this.fastPop(movePool, movePool.indexOf(moveid2));
+						if (moves.size + movePool.length <= this.maxMoveCount) return;
+					}
+				}
+			}
+			if (moveArrayA.includes(moveid1)) {
+				for (const moveid2 of moveArrayB) {
+					if (moveid1 !== moveid2 && movePool.includes(moveid2)) {
+						this.fastPop(movePool, movePool.indexOf(moveid2));
+						if (moves.size + movePool.length <= this.maxMoveCount) return;
+					}
+				}
+			}
+		}
+	}
+
+	// Adds a move to the moveset, returns the MoveCounter
+	addMove(
+		move: string,
+		moves: Set<string>,
+		types: string[],
+		abilities: Set<string>,
+		teamDetails: RandomTeamsTypes.TeamDetails,
+		species: Species,
+		isLead: boolean,
+		isDoubles: boolean,
+		movePool: string[],
+		preferredType: string,
+		role: string,
```

Comment:
Why not add a proper type for roles? That would catch typos made in the names of roles.

Reply:
I've created a type for roles